yarn-project/blob-sink/src/server.ts

```diff
@@ -0,0 +1,162 @@
+import { Blob } from '@aztec/foundation/blob';
+import { type Logger, createLogger } from '@aztec/foundation/log';
+import { type AztecKVStore } from '@aztec/kv-store';
+import { type TelemetryClient } from '@aztec/telemetry-client';
+import { NoopTelemetryClient } from '@aztec/telemetry-client/noop';
+
+import express, { type Express, type Request, type Response, json } from 'express';
+import { type Server } from 'http';
+import { z } from 'zod';
+
+import { type BlobStore, DiskBlobStore } from './blobstore/index.js';
+import { MemoryBlobStore } from './blobstore/memory_blob_store.js';
+import { type BlobSinkConfig } from './config.js';
+import { BlobSinkMetrics } from './metrics.js';
+import { type PostBlobSidecarRequest, blockIdSchema } from './types/api.js';
+import { BlobWithIndex } from './types/index.js';
+
+/**
+ * Example usage:
+ * const service = new BlobSinkService({ port: 5052 });
+ * await service.start();
+ * ... later ...
+ * await service.stop();
+ */
+export class BlobSinkServer {
+  public readonly port: number;
+
+  private app: Express;
+  private server: Server | null = null;
+  private blobStore: BlobStore;
+  private metrics: BlobSinkMetrics;
+  private log: Logger = createLogger('aztec:blob-sink');
+
+  constructor(config?: BlobSinkConfig, store?: AztecKVStore, telemetry: TelemetryClient = new NoopTelemetryClient()) {
+    this.port = config?.port ?? 5052; // 5052 is beacon chain default http port
+    this.app = express();
+
+    // Setup middleware
+    this.app.use(json({ limit: '1mb' })); // Increase the limit to allow for a blob to be sent
+
+    this.metrics = new BlobSinkMetrics(telemetry);
+
+    this.blobStore = store === undefined ? new MemoryBlobStore() : new DiskBlobStore(store);
+
+    // Setup routes
+    this.setupRoutes();
+  }
+
+  private setupRoutes() {
+    this.app.get('/eth/v1/beacon/blob_sidecars/:block_id', this.handleBlobSidecar.bind(this));
+    this.app.post('/blob_sidecar', this.handlePostBlobSidecar.bind(this));
+  }
+
+  private async handleBlobSidecar(req: Request, res: Response) {
+    // eslint-disable-next-line camelcase
+    const { block_id } = req.params;
+
+    try {
+      // eslint-disable-next-line camelcase
+      const parsedBlockId = blockIdSchema.parse(block_id);
```

Comment:
As mentioned, think there could be a bit of confusion around the block_id here as it is said to follow the same api but it don't.